src/ai/backend/agent/kernel.py

```diff
@@ -659,20 +666,30 @@ async def get_repl_in_addr(self) -> str:
     async def get_repl_out_addr(self) -> str:
         raise NotImplementedError
 
+    async def _close_task(self, task: Optional[asyncio.Task]) -> None:
```

Comment:
I'd suggest to separate this method, as it can be used generally for other cases.